test/apps/iodemo/io_demo.cc

```diff
@@ -832,132 +889,182 @@ class DemoClient : public P2pDemoCommon {
+            int active_index = IoDemoRandom::rand(size_t(0),
+                                                  _active_servers.size() - 1);
+            size_t server_index = _active_servers[active_index];
+            assert(_server_info[server_index].conn != NULL);
+
+            io_op_t op = get_op();
             size_t size;
             switch (op) {
             case IO_READ:
-                size = do_io_read(conn[conn_num], sn);
+                size = do_io_read(server_index, sn);
                 break;
             case IO_WRITE:
-                size = do_io_write(conn[conn_num], sn);
+                size = do_io_write(server_index, sn);
                 break;
             default:
                 abort();
             }
 
-            info[op].total_bytes += size;
-            info[op].num_iters++;
+            op_info[op].total_bytes += size;
+            op_info[op].num_iters++;
 
             if (((total_iter % 10) == 0) && (total_iter > total_prev_iter)) {
+                /* Print performance every 1 second */
                 double curr_time = get_time();
                 if (curr_time >= (prev_time + 1.0)) {
-                    if (!wait_for_responses(0)) {
+                    wait_for_responses(0);
+                    if (_status != OK) {
                         break;
                     }
 
                     report_performance(total_iter - total_prev_iter,
-                                       curr_time - prev_time, info);
+                                       curr_time - prev_time, op_info);
                     total_prev_iter = total_iter;
                     prev_time       = curr_time;
-
-                    check_time_limit(curr_time);
                 }
             }
 
             ++total_iter;
             ++sn;
         }
 
-        if (wait_for_responses(0)) {
+        wait_for_responses(0);
+        if (_status == OK) {
             double curr_time = get_time();
             report_performance(total_iter - total_prev_iter,
-                               curr_time - prev_time, info);
-            check_time_limit(curr_time);
+                               curr_time - prev_time, op_info);
         }
 
-        for (size_t i = 0; i < conn.size(); i++) {
-            LOG << "Disconnecting from server " << i;
-            delete conn[i];
-        }
-        return (_status == OK) || (_status == RUNTIME_EXCEEDED);
-    }
-
-    // returns true if has to stop the connection retries
-    bool update_retry() {
-        _status = OK;
-        check_time_limit(get_time());
-        if (_status == RUNTIME_EXCEEDED) {
-            /* the run-time of the application has been exhausted */
-            return true;
-        }
-
-        if (++_retry >= opts().client_retries) {
-            /* client failed all retries */
-            _status = CONN_RETRIES_EXCEEDED;
-            return true;
+        for (int server_index = 0; server_index < _server_info.size();
```

Comment:
why changed to `int`?

Reply:
for consistency since server_info is defined everywhere else as int

Reply:
make server_index size_t always

Reply:
Changed from int to size_t.